src/internal.c

```diff
@@ -7592,6 +7674,34 @@ static INLINE void CopyNameList(byte* buf, word32* idx,
     *idx = begin;
 }
 
+
+static word32 BuildNameList(char* buf, word32 bufSz,
+        const byte* src, word32 srcSz)
+{
+    const char* name;
+    word32 nameSz, idx;
+
+    idx = 0;
+    do {
+        name = IdToName(*src);
+        nameSz = (word32)WSTRLEN(name);
+
+        if (nameSz + 1 + idx > bufSz) {
+            break;
```

Comment:
This will truncate the name list silently, I think. Should we change this function's return type to int and return a negative value in this case? Then wherever we call BuildNameList we check that the return isn't negative before proceeding.